### ISSUE TITLE
[UPSTREAM_BUG] tags not being feteched in in checkout v4

### ISSUE THREAD

**Issue by rickstaa** [rickstaa/action-create-tag#41]
**Bug Description:**

A bug has been identified in version 4 of the [actions/checkout](https://github.com/actions/checkout) GitHub Action, where the `fetch-tags` argument set to `true` fails to fetch tags. This issue is documented on [GitHub](https://github.com/actions/checkout/issues/1471).

**Steps to Reproduce:**

1. Create a new repository.
2. Add the following action file in the `.github/workflows` folder:

```yml
name: Create/update tag
on:
  workflow_dispatch:
jobs:
  create-tag:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-tags: true
      - uses: rickstaa/action-create-tag@v1
        id: "tag_create"
        with:
          tag: "latest"
          tag_exists_error: false
          message: "Latest release"

      # Print result using the env variable.
      -  run: |
          echo "Tag already present: ${{ env.TAG_EXISTS }}"

      # Print result using the action output.
      - run: |
          echo "Tag already present: ${{ steps.tag_create.outputs.tag_exists }}"
```

4. Push the `latest` tag to the branch to ensure its existence.
6. Run the action using the GitHub Action interface.
7. Observe that the tag is not found, and a `hint: Updates were rejected because the tag already exists in the remote.` error is thrown.

**Expected Behavior:**

As per the [actions/checkout documentation](https://github.com/actions/checkout?tab=readme-ov-file#usage), the provided action recipe should fetch tags, and the tag should be recognized as already present.

**Solution:**

As a temporary solution, update the README to instruct users to use the `fetch-depth: 0` argument to fetch all tags until [GitHub issue #1471](https://github.com/actions/checkout/issues/1471) is resolved. Once fixed, the `fetch-tags: true` argument can be used again.

**Comment by rickstaa** [comment]
This will be fixed once #40 gets merged. Down the road, we can smoothly transition to the `fetch-tags` argument.

**Comment by rickstaa** [comment]
Closing this for now. I will update the documentation when https://github.com/actions/checkout/issues/1471 is fixed on the upstream.
